### ISSUE TITLE
Building wheel for pyobjc-framework-ScreenCaptureKit (setup.py) ... error

### ISSUE THREAD

**Issue by wildcolor** [ronaldoussoren/pyobjc#524]
**Describe the bug**
Not entirely sure if I could ask here. But this error stops my new program from moving ahead for quite a few days.

I am using MacOS 12.4 Monterey. I am tring to install with pip install pyobjc-framework-ScreenCaptureKit. But failed. I've got the following error:

Collecting pyobjc-framework-ScreenCaptureKit
  Using cached pyobjc-framework-ScreenCaptureKit-9.0.1.tar.gz (29 kB)
Requirement already satisfied: pyobjc-core>=9.0.1 in ./lib/python3.8/site-packages (from pyobjc-framework-ScreenCaptureKit) (9.0.1)
Requirement already satisfied: pyobjc-framework-CoreMedia>=9.0.1 in ./lib/python3.8/site-packages (from pyobjc-framework-ScreenCaptureKit) (9.0.1)
Requirement already satisfied: pyobjc-framework-Cocoa>=9.0.1 in ./lib/python3.8/site-packages (from pyobjc-framework-CoreMedia>=9.0.1->pyobjc-framework-ScreenCaptureKit) (9.0.1)
Building wheels for collected packages: pyobjc-framework-ScreenCaptureKit
  Building wheel for pyobjc-framework-ScreenCaptureKit (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/setup.py'"'"'; __file__='"'"'/private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-wheel-15qki0q8
       cwd: /private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/
  Complete output (25 lines):
  running bdist_wheel
  running build
  running build_py
  overriding build_packages to copy PyObjCTest
  creating build
  creating build/lib.macosx-10.14.6-x86_64-3.8
  creating build/lib.macosx-10.14.6-x86_64-3.8/ScreenCaptureKit
  copying Lib/ScreenCaptureKit/_metadata.py -> build/lib.macosx-10.14.6-x86_64-3.8/ScreenCaptureKit
  copying Lib/ScreenCaptureKit/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/ScreenCaptureKit
  creating build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
  copying PyObjCTest/test_scshareablecontent.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
  copying PyObjCTest/test_screencaputurekit.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
  copying PyObjCTest/test_scerror.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
  copying PyObjCTest/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
  copying PyObjCTest/test_scstream.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
  running build_ext
  building 'ScreenCaptureKit._ScreenCaptureKit' extension
  creating build/temp.macosx-10.14.6-x86_64-3.8
  creating build/temp.macosx-10.14.6-x86_64-3.8/Modules
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -I/Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/include -I/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8 -c Modules/_ScreenCaptureKit.m -o build/temp.macosx-10.14.6-x86_64-3.8/Modules/_ScreenCaptureKit.o -Wno-deprecated-declarations -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -DPyObjC_BUILD_RELEASE=1201 -Werror
  Modules/_ScreenCaptureKit.m:7:9: fatal error: 'ScreenCaptureKit/ScreenCaptureKit.h' file not found
  #import <ScreenCaptureKit/ScreenCaptureKit.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyobjc-framework-ScreenCaptureKit
  Running setup.py clean for pyobjc-framework-ScreenCaptureKit
Failed to build pyobjc-framework-ScreenCaptureKit
Installing collected packages: pyobjc-framework-ScreenCaptureKit
    Running setup.py install for pyobjc-framework-ScreenCaptureKit ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/setup.py'"'"'; __file__='"'"'/private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-record-yajbsb8k/install-record.txt --single-version-externally-managed --compile --install-headers /Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/include/site/python3.8/pyobjc-framework-ScreenCaptureKit
         cwd: /private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/
    Complete output (25 lines):
    running install
    running build
    running build_py
    overriding build_packages to copy PyObjCTest
    creating build
    creating build/lib.macosx-10.14.6-x86_64-3.8
    creating build/lib.macosx-10.14.6-x86_64-3.8/ScreenCaptureKit
    copying Lib/ScreenCaptureKit/_metadata.py -> build/lib.macosx-10.14.6-x86_64-3.8/ScreenCaptureKit
    copying Lib/ScreenCaptureKit/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/ScreenCaptureKit
    creating build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
    copying PyObjCTest/test_scshareablecontent.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
    copying PyObjCTest/test_screencaputurekit.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
    copying PyObjCTest/test_scerror.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
    copying PyObjCTest/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
    copying PyObjCTest/test_scstream.py -> build/lib.macosx-10.14.6-x86_64-3.8/PyObjCTest
    running build_ext
    building 'ScreenCaptureKit._ScreenCaptureKit' extension
    creating build/temp.macosx-10.14.6-x86_64-3.8
    creating build/temp.macosx-10.14.6-x86_64-3.8/Modules
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -I/Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/include -I/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8 -c Modules/_ScreenCaptureKit.m -o build/temp.macosx-10.14.6-x86_64-3.8/Modules/_ScreenCaptureKit.o -Wno-deprecated-declarations -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -DPyObjC_BUILD_RELEASE=1201 -Werror
    Modules/_ScreenCaptureKit.m:7:9: fatal error: 'ScreenCaptureKit/ScreenCaptureKit.h' file not found
    #import <ScreenCaptureKit/ScreenCaptureKit.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    error: command 'clang' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/setup.py'"'"'; __file__='"'"'/private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-install-hnei85xs/pyobjc-framework-screencapturekit/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/sh/7lm7ssmx6qg3p2hp9rsnsxkw0000gn/T/pip-record-yajbsb8k/install-record.txt --single-version-externally-managed --compile --install-headers /Users/wei/wei_wk_prog/chat_gpt_progs/paddlenlp/include/site/python3.8/pyobjc-framework-ScreenCaptureKit Check the logs for full command output.

**Platform information**

* Python version 3.8.9 (in Virtualenv)
* How was python installed (Virtual environment)
* macOS version 12.4 Monterey
I have Xcode installed. I assume Clang (which is needed for legacy build if no wheel package is installed) is installed with Xcode?
I also have wheel (pip install wheel) installed in the same virtual environment.

**Additional context**
I searched the wheel files and found .whl for MacOS 9, 10, 11, but failed to see wheel files for MacOS 12. Is this the problem? How can I deal with it?

**Comment by ronaldoussoren** [comment]
It looks like you have to update the developer tools, the build is using the macOS 12.1 SDK while ScreenCaptureKit.framework was introduced in macOS 12.3.

**Comment by wildcolor** [comment]
@ronaldoussoren Thanks for the kind reply.

Just to report the end result:

I tried to update everything, including the commandlinetools (13.4) , Xcode (to 14.2) and MacOS (to 12.6). But still failed to install the ScreenCaptureKit. But the other windows machine didn't have any problem for the installation. 

And today, I opened a new tab using 'homebrew'. I started the virtualenv with both ()(), which means with both 'the project virtualenv' and 'base - which is a conda virtualenv'. And this time, when I run pip install <package>. I just went fine.... Oh, I also had my vpn shadowsocks on global.

Not exactly sure, what was the problem. But just report everything here, maybe it would help other people.
